lnbits/commands.py

```diff
@@ -120,10 +145,367 @@ async def load_disabled_extension_list() -> None:
+            click.echo(f"  - {ext.id} ({ext.installed_release.version})")
+
+    _run_async(wrap)
+
+
+@extensions.command("upgrade")
+@click.argument("extension", required=False)
+@click.option("-a", "--all", is_flag=True, help="Upgrade all extensions.")
+@click.option(
+    "-i", "--repo-index", help="Select the index of the repository to be used."
+)
+@click.option(
+    "-s", "--source-repo", help="Provide the repository URL to be used for upgrading."
+)
+@click.option(
+    "-u",
+    "--url",
+    help="Use this option to update a running server. Eg: 'http://localhost:5000'.",
+)
+def extensions_upgrade(
+    extension: Optional[str] = None,
+    all: Optional[bool] = False,
+    repo_index: Optional[str] = None,
+    source_repo: Optional[str] = None,
+    url: Optional[str] = None,
+):
+    """Upgrade extensions"""
+    if not extension and not all:
+        click.echo("Extension ID is required.")
+        click.echo("Or specify the '--all' flag to upgrade all extensions")
+        return
+    if extension and all:
+        click.echo("Only one of extension ID or the '--all' flag must be specified")
+        return
+    if url and not _is_url(url):
+        click.echo(f"Invalid '--url' option value: {url}")
+        return
+
+    async def wrap():
+        await check_admin_settings()
+        if not await _can_run_operation(url):
+            return
+
+        if extension:
+            await upgrade_extension(extension, repo_index, source_repo, url)
+            return
+
+        click.echo("Upgrading all extensions...")
+        installed_extensions = await get_installed_extensions()
+        upgraded_extensions = []
+        for e in installed_extensions:
+            try:
+                click.echo(f"""{"="*50} {e.id} {"="*(50-len(e.id))} """)
+                success, msg = await upgrade_extension(
+                    e.id, repo_index, source_repo, url
+                )
+                if version:
+                    upgraded_extensions.append(
+                        {"id": e.id, "success": success, "message": msg}
+                    )
+            except Exception as ex:
+                click.echo(f"Failed to install extension '{e.id}': {ex}")
+
+        if len(upgraded_extensions) == 0:
+            click.echo("No extension was upgraded.")
+        else:
+            for u in sorted(upgraded_extensions, key=lambda d: d["id"]):
+                status = "upgraded to  " if u["success"] else "not upgraded "
+                click.echo(
+                    f"""'{u["id"]}' {" "*(20-len(u["id"]))}"""
+                    + f""" - {status}: '{u["message"]}'"""
+                )
+
+    _run_async(wrap)
+    return
+
+
+@extensions.command("install")
+@click.argument("extension")
+@click.option("--repo-index", help="Select the index of the repository to be used.")
+@click.option(
+    "--source-repo", help="Provide the repository URL to be used for installing."
+)
```

Comment:
provide short-options too (same as upgrade command)

```suggestion
@click.option("-i", "--repo-index", help="Select the index of the repository to be used.")
@click.option(
    "-s", "--source-repo", help="Provide the repository URL to be used for installing."
)
```